base/methodshow.jl

```diff
@@ -205,24 +205,44 @@ function show(io::IO, m::Method)
         print(io, m.name, "(...) in ", m.module)
         return
     end
-    print(io, decls[1][2], "(")
-    join(
-        io,
-        String[isempty(d[2]) ? d[1] : string(d[1], "::", d[2]) for d in decls[2:end]],
-        ", ",
-        ", ",
-    )
+    printstyled(io, decls[1][2]; color=:blue)
+    print(io, "(")
+    for (d, i) in zip(decls[2:end], 2:length(decls))
+        if isempty(d[2])
+            printstyled(io, d[1]; color=:magenta)
+        else
+            printstyled(io, d[1]; color=:magenta)
+            typespec = split(d[2], "{"; limit=2)
+            if length(typespec) == 1
+                printstyled(io, "::", d[2]; color=:yellow)
+            else
+                printstyled(io, "::", typespec[1]; color=:yellow)
+                printstyled(io, "{", typespec[2]; color=247)
```

Comment:
I think that in general we try to only use the 16 standard colors in the REPL since they are the ones most widely supported.

Reply:
Does anything not support 256-color? I thought perhaps TTYs but I just checked my TTY and it seems to work there...
![image](https://user-images.githubusercontent.com/20903656/119238933-3f747b00-bb78-11eb-8536-db287355db86.png)